sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.institutions.controller.ts

```diff
@@ -258,21 +277,59 @@ export class ApplicationOfferingChangeRequestInstitutionsController extends Base
    * @param payload information to create the new request.
    * @returns newly change request id created.
    */
+  @ApiNotFoundResponse({
+    description: "Application not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Study period overlap or offering program year mismatch or offering intensity mismatch.",
+  })
+  @ApiUnauthorizedResponse({
+    description: "The location does not have access to the offering.",
+  })
   @Post()
   async createApplicationOfferingChangeRequest(
     @UserToken() userToken: IInstitutionUserToken,
     @Param("locationId", ParseIntPipe) locationId: number,
     @Body() payload: CreateApplicationOfferingChangeRequestAPIInDTO,
   ): Promise<PrimaryIdentifierAPIOutDTO> {
-    // TODO: Apply the same validations from PIR.
-    const applicationOfferingChangeRequest =
-      await this.applicationOfferingChangeRequestService.createRequest(
-        locationId,
+    try {
+      await this.applicationService.applicationOfferingValidation(
         payload.applicationId,
         payload.offeringId,
-        payload.reason,
-        userToken.userId,
+        locationId,
       );
-    return { id: applicationOfferingChangeRequest.id };
+      const applicationOfferingChangeRequest =
+        await this.applicationOfferingChangeRequestService.createRequest(
+          locationId,
+          payload.applicationId,
+          payload.offeringId,
+          payload.reason,
+          userToken.userId,
+        );
+      return { id: applicationOfferingChangeRequest.id };
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        switch (error.name) {
+          case APPLICATION_NOT_FOUND:
```

Comment:
So far, the `ApiProcessError` is being used, as far as I know, to help to display expected errors to API used in a friendly way.
We handle this in two different ways.
1 - Use `ApiProcessError` when it is an expected business scenario where an API consumer (UI) needs to inform the user.
2 - Regular API exception when the error can be caused only by a user manipulating the payload unexpectedly.
To put in the current context, `APPLICATION_NOT_FOUND`, `OFFERING_PROGRAM_YEAR_MISMATCH`, `OFFERING_DOES_NOT_BELONG_TO_LOCATION` would be regular API errors instead of `ApiProcessError`.
Open to talking about maybe changing the pattern to have them all always expressed as `ApiProcessError`.